### PR TITLE
Replace Relish RSpec links with rspec.info

### PR DESCRIPTION
**Pull request by pirj**
Unfortunately, Relish is no more.

See:
 - https://mattwynne.net/relish
 - https://github.com/rspec/relish-root-repo/issues/5#issuecomment-1446152910
 - https://mattwynne.net/new-beginning
 - https://github.com/rspec/rspec.github.io/pull/183

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [x] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: docs/modules/ROOT/pages/cops_rspec.adoc [hunk 1]
```diff
@@ -2879,7 +2879,7 @@ end
 
 * https://rspec.rubystyle.guide/#declare-constants
 * https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/LeakyConstantDeclaration
-* https://relishapp.com/rspec/rspec-mocks/docs/mutating-constants
+* https://rspec.info/features/3-12/rspec-mocks/mutating-constants
```
Comment (pirj) [inline docs/modules/ROOT/pages/cops_rspec.adoc:2882]: I couldn't find a way to refer to the latest docs.

File: docs/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -32,7 +32,7 @@ over
 calculator.compute(line_item).should == 5
 ----
 
-is a feature of RSpec itself – you can read about it in the https://relishapp.com/rspec/rspec-expectations/docs/syntax-configuration#disable-should-syntax[RSpec Documentation].
+is a feature of RSpec itself – you can read about it in the "Disable should syntax" section of https://rspec.info/features/3-12/rspec-expectations/syntax-configuration[RSpec Documentation].
```
Comment (pirj) [inline docs/modules/ROOT/pages/index.adoc:35]: There are no anchors to paragraphs in the new docs.

**Comment by bquorning** [comment]
Do you know who runs rspec.info? The [about page](http://rspec.info/about/) doesn’t really say.

**Comment by pirj** [comment]
It's a custom domain for GitHub Pages of https://rspec.github.io which is this repo https://github.com/rspec/rspec.github.io/

**Review by bquorning — APPROVED** [review]

**Pull request merged**